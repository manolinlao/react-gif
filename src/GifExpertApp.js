import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
import { ExampleUseEffect } from './pruebas/ExampleUseEffect';

export const GifExpertApp = () => {

  const [ categories, setCategories ] = useState( [''] );
 
  return (    
    <div>
      <h2>PRUEBAS DE REACT</h2>
      <ExampleUseEffect/>

      <hr/>
      <h2>GifExpertApp</h2>
      <hr/>
            
      <AddCategory setCategories = { setCategories }/>

      <ol>
        {
          categories.map( category => {
            return <GifGrid key={ category } category={ category }/>
          })
        }
      </ol>
    </div>
  )
}
