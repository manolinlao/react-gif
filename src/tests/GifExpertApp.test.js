import React from 'react';
import { shallow } from "enzyme";
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas <GifExpertApp>',()=>{

  // Probamos que matche con el snapshot  
  test('match con el snapshot',()=>{
    const wrapper = shallow( <GifExpertApp/>);
    expect( wrapper ).toMatchSnapshot();   
  });

  // Probamos que matche con el snapshot  
  // y que tenga un GifGrid
  test('match con el snapshot y tenga 1 GifGrid',()=>{
    const wrapper = shallow( <GifExpertApp/>);
    expect( wrapper ).toMatchSnapshot();   
    expect( wrapper.find( 'GifGrid').length ).toBe( 1 );
  });

  
})
