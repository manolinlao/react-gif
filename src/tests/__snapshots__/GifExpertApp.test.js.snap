// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifExpertApp> match con el snapshot 1`] = `
<div>
  <h2>
    PRUEBAS DE REACT
  </h2>
  <ExampleUseEffect />
  <hr />
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category=""
      key=""
    />
  </ol>
</div>
`;

exports[`Pruebas <GifExpertApp> match con el snapshot 2`] = `
<div>
  <h2>
    PRUEBAS DE REACT
  </h2>
  <ExampleUseEffect />
  <hr />
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category=""
      key=""
    />
  </ol>
</div>
`;

exports[`Pruebas <GifExpertApp> match con el snapshot y tenga 1 GifGrid 1`] = `
<div>
  <h2>
    PRUEBAS DE REACT
  </h2>
  <ExampleUseEffect />
  <hr />
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category=""
      key=""
    />
  </ol>
</div>
`;
